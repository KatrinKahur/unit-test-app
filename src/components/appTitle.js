import React from "react";

export const AppTitle = ({title}) => {
    return(
        <>
            <h1 className="toolNameAbrev">JUTAI</h1>
            <p className="toolName">Java Unit Testing with AI</p>
        </>
    )
}