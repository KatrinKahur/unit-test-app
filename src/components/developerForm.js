import React from "react";

/**
 * This component is the development form of JUTAI,
 * used for collecting the measurements taken when evaluating its prompts
 * @param onSubmit The function called when submitting the measurements
 * @param setLineCoverage The function that sets the value of the line coverage criteria
 * @param setBranchCoverage The function that sets the value of the branch coverage criteria
 * @param setPassedUnitTests The function that sets the value of the passed unit tests criterua
 * @returns {JSX.Element} The developer form
 */
export const DeveloperForm = ({onSubmit, setLineCoverage, setBranchCoverage, setPassedUnitTests}) => {
    return(
        <form className="testForm" id="result-form"
              onSubmit={onSubmit}>
            <div>
                <label className="inputSquare">
                    <p className="testText">Line Coverage</p>
                    <textarea type="text" name="name" className="inputBox" required="required"
                              onChange={(e) => setLineCoverage(e.target.value)}/>
                </label>
                <label className="inputSquare">
                    <p className="testText">Branch Coverage</p>
                    <textarea type="text" name="name" className="inputBox" required="required"
                              onChange={(e) => setBranchCoverage(e.target.value)}/>
                </label>
                <label className="inputSquare">
                    <p className="testText">Passed Unit Tests</p>
                    <textarea type="text" name="name" className="inputBox" required="required"
                              onChange={(e) => setPassedUnitTests(e.target.value)}/>
                </label>
            </div>
            <input type="submit" value="Submit" className="testSubmitButt"/>
        </form>
    )
}