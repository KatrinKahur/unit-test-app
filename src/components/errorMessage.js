import {Alert} from "react-bootstrap";
import React from "react";

export const ErrorMessage = ({message, show, setShow}) => {

    if(message === null){
        return null
    }

    if(show){
        return (
            <Alert data-testid="error-message" style={{position: "fixed", top:10, left: 0, right: 0, zIndex: 999}} variant="secondary" onClose={() => setShow(false)} dismissible>
                <Alert.Heading>Error!</Alert.Heading>
                <p>{message}</p>
            </Alert>
        );
    }
}