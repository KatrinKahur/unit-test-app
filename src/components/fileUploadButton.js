/**
 * This function represents a file upload button.
 * @param callback The function called when a file is selected.
 * @returns {JSX.Element} file upload button
 */
export const FileUploadButton = ({onChange}) => {
    return (
        <div className="inputButt">
            <input type="file"
                   id="file-input"
                   data-testid="file-input"
                   onChange={onChange}
                   required
            />
        </div>
    )
}