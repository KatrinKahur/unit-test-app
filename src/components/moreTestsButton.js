export const MoreTestsButton = ({onClick}) => {
    return(
        <button className="moreTestsButton" onClick={onClick}>Generate more</button>
    )
}