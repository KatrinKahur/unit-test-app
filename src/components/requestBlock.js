import SyntaxHighlighter from "react-syntax-highlighter";
import railscasts from "react-syntax-highlighter/dist/cjs/styles/hljs/railscasts";
import React from "react";

export const RequestBlock = ({javaClass}) => {
    return(
        <div className="requestBlockWrapper">
            <h3>Java class: </h3>
            <div className="requestBlock">
                <SyntaxHighlighter
                    data-testid="request-block"
                    wrapLines={true}
                    lineProps={{style: {whiteSpace: 'pre-wrap'}}}
                    language="java"
                    style={railscasts}>{javaClass}
                </SyntaxHighlighter>
            </div>
        </div>
    )
}