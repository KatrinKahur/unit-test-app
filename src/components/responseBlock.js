import {Spinner} from "react-bootstrap";
import SyntaxHighlighter from "react-syntax-highlighter";
import React from "react";
import {a11yLight} from "react-syntax-highlighter/dist/esm/styles/hljs";

export const ResponseBlock = ({loadingStatus, response}) => {
    return(
        <div className="responseBlockWrapper">
            <h3>Tests: </h3>
            <div className="responseBlock">
                {loadingStatus && <Spinner animation="border" variant="secondary" />}
                <SyntaxHighlighter
                    wrapLines={true}
                    lineProps={{style: {whiteSpace: 'pre-wrap'}}}
                    language="java"
                    style={a11yLight}>{response}
                </SyntaxHighlighter>
            </div>
        </div>
    )
}