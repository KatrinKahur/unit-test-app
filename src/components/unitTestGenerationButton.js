import React from "react";

export const UnitTestGenerationButton = () => {
    return (
        <input data-cy="test-generation-btn" type="submit" value="Generate unit tests!"
               className="testGeneratorButton" />
    )
}