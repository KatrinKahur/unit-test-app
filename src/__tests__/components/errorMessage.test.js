import '@testing-library/jest-dom';
import '@testing-library/user-event';
import {ErrorMessage} from "../../components/errorMessage";
import {cleanup, fireEvent, render, screen} from "@testing-library/react";

beforeEach(() => {
    cleanup();
})
describe('ErrorMessage', () => {
    it('should render correctly', () => {
        let showStatus = false;
        const setShow = (status) => {
            showStatus = status;
        }
        const testProps = {
            message: "Error message",
            show: true,
            setShow: setShow
        }

        const { asFragment } = render(<ErrorMessage {...testProps}/>)
        expect(asFragment()).toMatchSnapshot();
    });

    it('should appear when the show variable is set', () => {
        let showStatus = false;
        let errorMsg = "Error message";
        const setShow = (status) => {
            showStatus = status;
        }
        const testProps = {
            message: errorMsg,
            show: true,
            setShow: setShow
        }

        render(<ErrorMessage {...testProps} />)
        expect(screen.getByText(errorMsg)).toBeInTheDocument();

    });

    it('should not appear when the show variable is false', () => {
        let showStatus = false;
        let errorMsg = "Error message";
        const setShow = (status) => {
            showStatus = status;
        }
        const testProps = {
            message: errorMsg,
            show: false,
            setShow: setShow
        }
        render(<ErrorMessage {...testProps} />)
        expect(screen.queryByText(errorMsg)).toBeNull();
    });

    it('should disappear when pressed on', async () => {
        let showStatus = false;
        let errorMsg = "Error message";
        const setShow = (status) => {
            showStatus = status;
        }
        const testProps = {
            message: errorMsg,
            show: true,
            setShow: setShow
        }
        render(<ErrorMessage {...testProps} />)
        expect(screen.getByTestId('error-message')).toBeInTheDocument();

        await(async() => {
            fireEvent.click(screen.getByTestId('error-message'));
        })

        expect(screen.getByText(errorMsg)).toBeInTheDocument();

    })
})