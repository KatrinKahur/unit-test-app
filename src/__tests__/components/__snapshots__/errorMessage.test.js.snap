// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render correctly 1`] = `
<DocumentFragment>
  <div
    class="fade alert alert-danger alert-dismissible show"
    data-testid="error-message"
    role="alert"
    style="position: fixed; top: 10px; left: 0px; right: 0px; z-index: 999;"
  >
    <button
      aria-label="Close alert"
      class="btn-close"
      type="button"
    />
    <div
      class="alert-heading h4"
    >
      Error!
    </div>
    <p>
      Error message
    </p>
  </div>
</DocumentFragment>
`;
